genmc-sys/build.rs

```diff
@@ -0,0 +1,201 @@
+        .file("./src_cpp/MiriInterface.cpp")
+        .compile("genmc_interop");
+
+    // Link the Rust-C++ interface library generated by cxx_build:
+    println!("cargo::rustc-link-lib=static=genmc_interop");
+}
+
+/// Build the GenMC model checker library.
+/// Returns the path where cmake installs the model checker library and the config.h file.
+fn build_genmc_model_checker(genmc_path: &Path) -> PathBuf {
+    let cmakelists_path = genmc_path.join("CMakeLists.txt");
+
+    let mut config = cmake::Config::new(cmakelists_path);
+    config.profile(GENMC_CMAKE_PROFILE);
+    if ENABLE_GENMC_DEBUG {
+        config.define("GENMC_DEBUG", "ON");
+    }
+
+    // FIXME(GenMC,HACK): Required for unknown reasons on older cmake (version 3.22.1, works without this with version 3.31.6)
+    //              Without this, the files are written into the source directory by the cmake configure step, and then
+    //              the build step cannot find these files, because it correctly tries using the `target` directory.
+    let out_dir = std::env::var("OUT_DIR").unwrap();
+    let genmc_build_path: PathBuf = [&out_dir, "build"].into_iter().collect();
+    config.configure_arg(format!("-B {}", genmc_build_path.display()));
+
+    // Enable only the components of GenMC that we need:
+    config.define("BUILD_LLI", "OFF");
+    config.define("BUILD_INSTRUMENTATION", "OFF");
+    config.define("BUILD_MODEL_CHECKER", "ON");
+
+    let cmake_install_dir = config.build();
+
+    // Add the model checker library to be linked and the install directory where it is located:
+    println!("cargo::rustc-link-search=native={}", cmake_install_dir.display());
+    println!("cargo::rustc-link-lib=static={GENMC_MODEL_CHECKER}");
+
+    cmake_install_dir
+}
+
+fn main() {
+    // Select between local GenMC repo, or downloading GenMC from a specific commit.
+    let Ok(genmc_local_path) = PathBuf::from_str(GENMC_LOCAL_PATH);
+    let genmc_path = if genmc_local_path.exists() {
+        // If the local repository exists, cargo should watch it for changes:
+        // FIXME(genmc,cargo): We could always watch this path even if it doesn't (yet) exist, depending on how `https://github.com/rust-lang/cargo/issues/6003` is resolved.
+        //                     Adding it here means we don't rebuild if a user creates `GENMC_LOCAL_PATH`, which isn't ideal.
+        //                     Cargo currently always rebuilds if a watched directory doesn't exist, so we can only add it if it exists.
+        println!("cargo::rerun-if-changed={GENMC_LOCAL_PATH}");
+        genmc_local_path
+    } else if cfg!(feature = "download_genmc") {
+        downloading::download_genmc()
+    } else {
+        panic!("GenMC not found in path '{GENMC_LOCAL_PATH}', and downloading GenMC is disabled.");
+    };
+
+    // Build all required components:
+    let genmc_install_dir = build_genmc_model_checker(&genmc_path);
+    build_cxx_bridge(&genmc_path, &genmc_install_dir);
```

Comment:
This should only use `genmc_install_dir` since the headers should also be installed.